server/app/controllers/fusor/api/v21/subscriptions_controller.rb

```diff
@@ -13,6 +13,19 @@
 module Fusor
   class Api::V21::SubscriptionsController < Api::V2::BaseController
 
+    before_filter :check_content_type, :only => :upload
+
+    def check_content_type
+      # overwrite this method in V2::BaseController that
+      # renders error if request.content_type != "application/json"
+      # the upload method uses "multipart/form-data"
+    end
+
+    def upload
+      # placeholder only. TODO perform actually upload file
+      render :json => {params: params, request: @request}
```

Comment:
@jmrodri, here's a placeholder route and method for `POST fusor/api/v21/subscriptions/upload` and you will be see in the "network" console of your browser that you have all the info that you need.